release/release.go

```diff
@@ -37,41 +37,33 @@ type OS struct {
 	VersionID string `json:"version-id,omitempty"`
 }
 
+var (
+	versionSignaturePath = "/proc/version_signature"
+	apparmorSysPath      = "/sys/kernel/security/apparmor/"
+)
+
 // ForceDevMode returns true if the distribution doesn't implement required
 // security features for confinement and devmode is forced.
-func (os *OS) ForceDevMode() bool {
-	supported := []struct {
-		ID        string
-		VersionID string
-	}{
-		{"neon", ""},
-		{"ubuntu", ""},
-		{"ubuntu-core", ""},
-		// NOTE: elementary is "Elementary OS" but normalization
-		// applies
-		{"elementary", "0.4"},
-		// NOTE: mint uses "LinuxMint" (mixed capitalization)
-		// but this is normalized by readOSRelease.
-		{"linuxmint", "18.1"},
-		{"galliumos", "2.0"},
-		{"peppermint", "7.0"},
-		// NOTE: zorin is "Zorin OS" but normalization applies
-		{"zorin", "12"},
+func (o *OS) ForceDevMode() bool {
+	// Check if kernel signature contains "Ubuntu", currently only
+	// the Ubuntu kernels have all the required apparmor patches
+	// (but those are getting upstreamed so at some point we need
+	// to make this check smater)
+	versionSig, err := ioutil.ReadFile(versionSignaturePath)
+	if err != nil {
```

Comment:
If the file is there but we cannot read it then yes. If the file is just not there I'd not log anything.